fractions/jose/src/main/java/org/wildfly/swarm/jose/JoseConfiguration.java

```diff
@@ -0,0 +1,260 @@
+    }
+
+    public String keyEncryptionAlgorithm() {
+        return this.keyEncryptionAlgorithm.get();
+    }
+
+    public JoseConfiguration contentEncryptionAlgorithm(String algorithm) {
+        contentEncryptionAlgorithm.set(algorithm);
+        return this;
+    }
+
+    public String contentEncryptionAlgorithm() {
+        return this.contentEncryptionAlgorithm.get();
+    }
+    /**
+     * Keystore type.
+     */
+    @Configurable("swarm.jose.keystore.type")
+    @AttributeDocumentation("Keystore type")
+    private Defaultable<String> keystoreType = string(DEFAULT_KEYSTORE_TYPE);
+
+    /**
+     * Path to the keystore.
+     */
+    @Configurable("swarm.jose.keystore.path")
+    @AttributeDocumentation("Path to the keystore")
+    private Defaultable<String> keystorePath = string(DEFAULT_KEYSTORE_PATH);
+
+    /**
+     * Password for the keystore.
+     */
+    @Configurable("swarm.jose.keystore.password")
+    @AttributeDocumentation("Password to the keystore")
+    private Defaultable<String> keystorePassword = string(DEFAULT_KEYSTORE_PASSWORD);
+
+    /**
+     * Signature algorithm.
+     */
+    @Configurable("swarm.jose.signature.algorithm")
+    @AttributeDocumentation("Signature algorithm")
+    private Defaultable<String> signatureAlgorithm = string(DEFAULT_SIGNATURE_ALGORITHM);
+
+    /**
+     * Signature Format.
+     */
+    @Configurable("swarm.jose.signature.format")
+    @AttributeDocumentation("Compact or JSON JWS format, support for JSON is optional")
+    private Defaultable<String> signatureFormat = string(DEFAULT_JOSE_FORMAT.name());
+
+    /**
+     * Signature Data Encoding.
+     */
+    @Configurable("swarm.jose.signature.data-encoding")
+    @AttributeDocumentation("JWS data encoding mode, true - base64url (default), false - clear text")
+    private Defaultable<Boolean> signatureDataEncoding = bool(DEFAULT_SIGNATURE_DATA_ENCODING);
+
+    /**
+     * Signature Detached Data.
+     */
+    @Configurable("swarm.jose.signature.data-detached")
+    @AttributeDocumentation("JWS data detached mode, true - the data is in the JWS payload (default), false - outside")
+    private Defaultable<Boolean> signatureDataDetached = bool(DEFAULT_SIGNATURE_DATA_DETACHED);
+
+    /**
+     * Password for the signature key.
+     */
+    @Configurable("swarm.jose.signature.key.password")
+    @AttributeDocumentation("Password to the signature private key")
+    private Defaultable<String> signatureKeyPassword = string(DEFAULT_KEY_PASSWORD);
+
+    /**
+     * Alias to the signature key entry in the keystore.
+     */
+    @Configurable("swarm.jose.signature.key.alias")
+    @AttributeDocumentation("Alias to the signature key entry in the keystore")
+    private Defaultable<String> signatureKeyAlias = string(DEFAULT_KEY_ALIAS);
+
+    /**
+     * Encryption Format.
+     */
+    @Configurable("swarm.jose.encryption.format")
+    @AttributeDocumentation("Compact or JSON JWE format, support for JSON is optional")
```

Comment:
What are the possible values?

Reply:
Also, not quite sure as a user what "support for JSON is optional" means. What guarantees do I have?

Reply:
Same as with the signature, Jose4j does not support this format for the encryption, Jose4j is primed for the compact sequences which are use in OAuth2/OIDC